ARKitExample/ViewController.swift

```diff
@@ -0,0 +1,204 @@
+     * This pretty fast but still wasteful, it would be nicer to wrap the CGImage and use its CGDataProvider directly.
+     **/
+    func copyPixelbufferFromCGImageProvider(image: CGImage) -> CVPixelBuffer {
+        let dataProvider: CGDataProvider? = image.dataProvider
+        let data: CFData? = dataProvider?.data
+        let baseAddress = CFDataGetBytePtr(data!)
+
+        /**
+         * We own the copied CFData which will back the CVPixelBuffer, thus the data's lifetime is bound to the buffer.
+         * We will use a CVPixelBufferReleaseBytesCallback callback in order to release the CFData when the buffer dies.
+         **/
+        let unmanagedData = Unmanaged<CFData>.passRetained(data!)
+        var pixelBuffer: CVPixelBuffer? = nil
+        let status = CVPixelBufferCreateWithBytes(nil,
+                                                  image.width,
+                                                  image.height,
+                                                  TVIPixelFormat.format32BGRA.rawValue,
+                                                  UnsafeMutableRawPointer( mutating: baseAddress!),
+                                                  image.bytesPerRow,
+                                                  { releaseContext, baseAddress in
+                                                    let contextData = Unmanaged<CFData>.fromOpaque(releaseContext!)
+                                                    contextData.release() },
+                                                  unmanagedData.toOpaque(),
+                                                  nil,
+                                                  &pixelBuffer)
+
+        if (status != kCVReturnSuccess) {
+            return nil as CVPixelBuffer!;
+        }
+
+        return pixelBuffer!
+    }
+}
+
+// MARK: ARSCNViewDelegate
+extension ViewController: ARSCNViewDelegate {
+    func session(_ session: ARSession, didFailWithError error: Error) {
+        // Present an error message to the user
+        print("didFailWithError \(error)")
+    }
+
+    func sessionWasInterrupted(_ session: ARSession) {
+        // Inform the user that the session has been interrupted, for example, by presenting an overlay
+        print("ARSession was interrupted, disabling the VideoTrack.")
+        self.videoTrack?.isEnabled = false
+    }
+
+    func sessionInterruptionEnded(_ session: ARSession) {
+        // Reset tracking and/or remove existing anchors if consistent tracking is required
+        print("ARSession interruption ended, enabling the VideoTrack.")
+        self.videoTrack?.isEnabled = true
+    }
+}
+
+// MARK: TVIRoomDelegate
+extension ViewController: TVIRoomDelegate {
+    func didConnect(to room: TVIRoom) {
+        print("Connected to Room /(room.name).")
+    }
+
+    func room(_ room: TVIRoom, didFailToConnectWithError error: Error) {
+        print("Failed to connect to a Room: \(error).")
+    }
```

Comment:
This is probably worthy of a UIAlertView since developers may forget to add a token.

Reply:
Yes, I agree. We should definitely display something to alert the user.

Reply:
👍 I think this is the minimal UI we can get away with before merging the example.